web/src/main/java/org/cbioportal/web/util/InvolvedCancerStudyExtractorInterceptor.java

```diff
@@ -137,6 +137,64 @@
         return true;
     }
 
+    private boolean extractAttributesFromClinicalDataBinCountFilter(HttpServletRequest request) {
+        try {
+            ClinicalDataBinCountFilter clinicalDataBinCountFilter = objectMapper.readValue(request.getReader(),
+                    ClinicalDataBinCountFilter.class);
+            LOG.debug("extracted clinicalDataBinCountFilter: " + clinicalDataBinCountFilter.toString());
+            LOG.debug("setting interceptedClinicalDataBinCountFilter to " + clinicalDataBinCountFilter);
+            request.setAttribute("interceptedClinicalDataBinCountFilter", clinicalDataBinCountFilter);
+            if (cacheMapUtil.hasCacheEnabled()) {
+                Collection<String> cancerStudyIdCollection = extractCancerStudyIdsFromClinicalDataBinCountFilter(
+                        clinicalDataBinCountFilter);
+                LOG.debug("setting involvedCancerStudies to " + cancerStudyIdCollection);
+                request.setAttribute("involvedCancerStudies", cancerStudyIdCollection);
+            }
+        } catch (Exception e) {
+            LOG.error("exception thrown during extraction of clinicalDataBinCountFilter: " + e);
+            return false;
+        }
+        return true;
+    }
+
+    private boolean extractAttributesFromClinicalDataCountFilter(HttpServletRequest request) {
+        try {
+            ClinicalDataCountFilter clinicalDataCountFilter = objectMapper.readValue(request.getReader(),
+                    ClinicalDataCountFilter.class);
+            LOG.debug("extracted clinicalDataBinCountFilter: " + clinicalDataCountFilter.toString());
+            LOG.debug("setting interceptedClinicalDataCountFilter to " + clinicalDataCountFilter);
+            request.setAttribute("interceptedClinicalDataCountFilter", clinicalDataCountFilter);
+            if (cacheMapUtil.hasCacheEnabled()) {
+                Collection<String> cancerStudyIdCollection = extractCancerStudyIdsFromClinicalDataCountFilter(
+                        clinicalDataCountFilter);
+                LOG.debug("setting involvedCancerStudies to " + cancerStudyIdCollection);
+                request.setAttribute("involvedCancerStudies", cancerStudyIdCollection);
+            }
+        } catch (Exception e) {
+            LOG.error("exception thrown during extraction of clinicalDataBinCountFilter: " + e);
+            return false;
+        }
+        return true;
+    }
+
+    private boolean extractAttributesFromStudyViewFilter(HttpServletRequest request) {
```

Comment:
Another copy is on line 504 .. I do notice a bug in both of these ... the Log.error() reports clinicalDataBinCountFilter, but this (and the previous function) are parsing different types. But a bugfix should be in a separate PR.

Reply:
didn't fix this - we can address it in a separate PR as suggested